pkg/skaffold/deploy/status_check.go

```diff
@@ -74,23 +69,29 @@ func StatusCheck(ctx context.Context, defaultLabeller *DefaultLabeller, runCtx *
 	return getSkaffoldDeployStatus(syncMap)
 }
 
-func getDeployments(client kubernetes.Interface, ns string, l *DefaultLabeller) (map[string]int32, error) {
+func getDeployments(client kubernetes.Interface, ns string, l *DefaultLabeller, deadlineDuration time.Duration) (map[string]time.Duration, error) {
 	deps, err := client.AppsV1().Deployments(ns).List(metav1.ListOptions{
 		LabelSelector: l.K8sManagedByLabelKeyValueString(),
 	})
 	if err != nil {
 		return nil, errors.Wrap(err, "could not fetch deployments")
 	}
 
-	depMap := map[string]int32{}
+	depMap := map[string]time.Duration{}
 
 	for _, d := range deps.Items {
-		var deadline int32
-		if d.Spec.ProgressDeadlineSeconds == nil {
-			logrus.Debugf("no progressDeadlineSeconds config found for deployment %s. Setting deadline to %d seconds", d.Name, defaultStatusCheckDeadlineInSeconds)
-			deadline = defaultStatusCheckDeadlineInSeconds
-		} else {
-			deadline = *d.Spec.ProgressDeadlineSeconds
+		var deadline time.Duration
+		switch d.Spec.ProgressDeadlineSeconds {
```

Comment:
this seems like a weird way to check `if d.Spec.ProgressDeadlineSeconds == nil`. do you think the debug logs here are useful? if not, this might be a little cleaner:

```golang
if d.Spec.ProgressDeadlineSeconds == nil || 
	(*d.Spec.ProgressDeadlineSeconds > int32(deadlineDuration.Seconds())) {
	deadline = deadlineDuration
} else {
	deadline = time.Duration(*d.Spec.ProgressDeadlineSeconds) * time.Second
}
```